controllers/login.controller.es6

```diff
@@ -20,17 +21,49 @@ export default class LoginController {
     this.alertGroup.registerUpdateListener(alerts => {
       this.alerts = alerts;
     });
+    this.ledgerStatus = 'Not connected';
+    this.connectLedger();
     Alerts.registerGroup(this.alertGroup);
   }
 
   broadcastShowDashboardIntent() {
+    this.ledgerApi.clearDeviceListeners();
     this.IntentBroadcast.sendBroadcast(
       new Intent(
         Intent.TYPES.SHOW_DASHBOARD
       )
     );
   }
 
+  connectLedger() {
+    this.ledgerApi = new StellarLedger.Api(new StellarLedger.comm(3));
+    let self = this;
+    this.ledgerApi.addDeviceListener(function(status, msg) {
```

Comment:
If you change it to arrow function (`(status, msg) => {`) you don't have to introduce `self` variable because arrows share the same lexical `this` as their surrounding code.